### PR TITLE
Image viewer brightness control

### PR DESCRIPTION
**Pull request by dagewa**
There may be different opinions about this, but I find the `dials.image_viewer` brightness to usually start too dark, to step too much on clicking in the scroll bar, and all the useful range being up the left end of the scale. This PR adjusts defaults to improve things a little. I welcome other views.

### CODE REVIEW

**Comment by dagewa** [comment]
I think we can do better. I'll try a non-linear function when I get some time.

**Comment by ndevenish** [comment]
image

**Comment by ndevenish** [comment]
This seems to lose a lot of the controllability and maximum contrast. The selector (running via NX to diamond) seems only to "jump" when dragging between 1, 2, 6, 10, 14, 18, 22, 26, 30 making it darker, and a similar but slightly different range when dragging lighter. Fast-dragging frustratingly seems to allow spaces in-between.

In addition, it seems to lose a lot of the maximum contrast end:
image

I feel that it should definitely allow you to make it "Darker" than is useful, because then you can see the extreme ends and drag somewhere in the middle.

**Comment by dagewa** [comment]
Thanks for the comments. I found the dark end of the scale fairly useless, so I cut it back earlier, but I can certainly extend it darker again. I'm not sure about the jumping behaviour. I don't think I saw that locally, but I'll investigate

**Comment by ndevenish** [comment]
Actually, the behaviour that I'm observing is that it's not moving until I'm off the edge of the handle. This might be a specific linux/wx interaction problem


https://user-images.githubusercontent.com/529963/124601163-6bfe0100-de5f-11eb-84ef-0053a2ad4dd0.mp4

**Comment by dagewa** [comment]
Ah right, actually I see the same. I had taken to clicking in the bar either side of the button, which steps by +/- 2

File: command_line/image_viewer.py [hunk 1]
```diff
@@ -28,7 +28,7 @@
 
 phil_scope = iotbx.phil.parse(
     """\
-brightness = 100
+brightness = 20
```
Comment (graeme-winter) [inline command_line/image_viewer.py:31]: Approve of this change (though personally I would have set 10)

File: newsfragments/1774.feature [hunk 2]
```diff
@@ -0,0 +1 @@
+``dials.image_viewer``: adjust brightness control for better defaults.
```
Comment (graeme-winter) [inline newsfragments/1774.feature:1]: Better is an opinion, would say they are certainly better for photon counting detectors

**Review by graeme-winter — CHANGES_REQUESTED** [review]
I am not certain I fully agree with the changes in here - while the "slider" is integral it becomes very hard to find good values with the changes in this PR for a stack of 10 images

The newly nonlinear nature of the scaling is to me troublesome... but I am not sure what to suggest in its place

Please could you try it out yourself with a stack of 10 images then see how you feel about the use of dynamic range (hint: only the left hand extreme of the slider is useful)

**Comment by graeme-winter** [comment]
> I am not certain I fully agree with the changes in here - while the "slider" is integral it becomes very hard to find good values with the changes in this PR for a stack of 10 images
> 
> The newly nonlinear nature of the scaling is to me troublesome... but I am not sure what to suggest in its place
> 
> Please could you try it out yourself with a stack of 10 images then see how you feel about the use of dynamic range (hint: only the left hand extreme of the slider is useful)

In fairness, this accusation should also be levelled at the existing implementation

**Comment by dagewa** [comment]
It still looks a lot better to me :thinking:

**Main `stack=1`**
main-stack-1

**Branch `stack=1`**
branch-stack-1

Neither very much use, but I can at least start to see diffraction spots on the branch

**Main `stack=10`**
main-stack-10

**Branch `stack=10`**
branch-stack-10

The branch behaviour is _much_ better in this case. On main it's just a dark mess. On the branch I get a clean diffraction image.

**Comment by dagewa** [comment]
@graeme-winter I've been discussing this with @ndevenish. I'm not able to reproduce anything concerning with the example I tried, so could you please provide an example to illuminate your request for changes? Thanks

**Comment by ndevenish** [comment]
Since there's lack of consensus on this now we'll leave for this release.

**Pull request closed without merging**